--- nextcloud.desktop/src/gui/CMakeLists.txt.orig	2021-07-12 17:16:57.918459069 +0200
+++ nextcloud.desktop/src/gui/CMakeLists.txt	2021-07-12 17:17:04.162536983 +0200
@@ -257,7 +257,7 @@
   set(icon_output_name "${size}-${icon_name_wle}.png")
   message(STATUS "Generate ${icon_output_name}")
   execute_process(COMMAND
-    "${INKSCAPE}" -w ${size} -h ${size} "${icon_path}" -o "${icon_output_name}"
+    "${INKSCAPE}" -w ${size} -h ${size} "${icon_path}" -e "${icon_output_name}"
     WORKING_DIRECTORY "${icon_name_dir}"
     RESULT_VARIABLE
     INKSCAPE_SIDEBAR_ERROR
