--- nextcloud-client-2.5.3.orig/src/3rdparty/kmessagewidget/kmessagewidget.cpp 2019-07-26 18:40:34.949349387 +0000
+++ nextcloud-client-2.5.3/src/3rdparty/kmessagewidget/kmessagewidget.cpp      2019-07-26 18:41:39.866478051 +0000
@@ -105,6 +105,9 @@
     q->setMessageType(KMessageWidget::Information);
 }

+template <typename T>
+constexpr typename std::add_const<T>::type &qAsConst(T &t) noexcept { return t; }
+
 void KMessageWidgetPrivate::createLayout()
 {
     delete content->layout();
--- a/src/libsync/syncengine.cpp
+++ b/src/libsync/syncengine.cpp
@@ -326,6 +326,11 @@ void SyncEngine::deleteStaleErrorBlacklistEntries(const SyncFileItemVector &sync
     _journal->deleteStaleErrorBlacklistEntries(blacklist_file_paths);
 }

+#if (QT_VERSION < 0x050600)
+template <typename T>
+constexpr typename std::add_const<T>::type &qAsConst(T &t) noexcept { return t; }
+#endif
+
 void SyncEngine::conflictRecordMaintenance()
 {
     // Remove stale conflict entries from the database
